Enterprise-CMCS/managed-care-review
# [MCR-2625] Limit State Users to One File Upload for Rate Certs

## Summary

#### Related issues
https://qmacbis.atlassian.net/browse/MCR-2625

#### To Do

- [x] Copy change for error message on Rate details page

#### Screenshots
<img width="452" alt="Screenshot 2023-09-27 at 11 03 07 AM" src="https://github.com/Enterprise-CMCS/managed-care-review/assets/67110378/4b637720-164b-4267-a493-6d7fa1d70664">


<img width="469" alt="Screenshot 2023-09-27 at 11 13 29 AM" src="https://github.com/Enterprise-CMCS/managed-care-review/assets/67110378/53b6f33a-e097-4c88-9d1f-e978b230cf4c">



#### Test cases covered

- `FileUpload` only allows 1 upload when its `multiple` prop is set to false
- Rate details page displays an error if the submission was unlocked and had multiple files uploaded
- Submit Review page displays an error if there are multiple rate cert files uploaded

## QA guidance

With the current code you can test locally that only one rate cert file can be uploaded. If you try to upload a second file after the first was added you should see that the first file was removed and replaced with the most recent addition 

To test that error messages appear on the Review Submit page and the Rate Details page (for an unlocked submission)

- Set `allowMultipleUploads` to true in `SingleRateCert.ts`
- As a state user add multiple files on the Rate details page for the rate cert field
- Set `allowMultipleUploads` back to false in `SingleRateCert.ts`
- On the summary page ensure that the previously uploaded files still appear
- Sign in as a CMS user and unlock the submission
- Sign as a state user and see the summary page, ensure the "Only one document allowed" error message appears above the file list
- Click edit to go the the unlocked Rate details page and ensure that if you click Continue without making any changes you see an "Only one document.." error message
